File: apps/mybookkeeper/frontend/src/shared/types/applicant/applicant-list-response.ts
import type { ApplicantSummary } from "./applicant-summary";

/**
 * Paginated envelope returned by GET /applicants — same shape as
 * ``InquiryListResponse``. ``has_more`` lets the frontend hide the
 * "Load more" button on the last page.
 */
export interface ApplicantListResponse {
  items: ApplicantSummary[];
  total: number;
  has_more: boolean;
}

File: apps/mybookkeeper/backend/app/schemas/applicants/applicant_list_response.py
"""Paginated envelope for GET /applicants — same shape as InquiryListResponse."""
from __future__ import annotations

from platform_shared.schemas.pagination import ListResponse

from app.schemas.applicants.applicant_summary import ApplicantSummary


class ApplicantListResponse(ListResponse[ApplicantSummary]):
    pass
